### PR TITLE
Rewards for decativated pool

### PR DESCRIPTION
**Pull request by AlexChetverov**

### CODE REVIEW

File: test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts [hunk 1]
```diff
@@ -233,3 +251,52 @@ test("Given a pool with 2 users with activated rewards WHEN more than one period
...
+      testUser1.getAsset(MGA_ASSET_ID)!.amountBefore.free!
+    );
+
+  await promotePool(sudo.keyRingPair, liqIdPromPool2, 20);
+
+  await waitForRewards(testUser1, liqIdPromPool2);
+
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+
+  await claimRewardsAll(testUser1, liqIdPromPool2).then((result) => {
```
Comment (goncer) [inline test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts:287]: either use the regular claim reward or generate a user on every test. here you are claiming rewards for all the pools; perhaps previous tests already generated some.

File: test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts [hunk 2]
```diff
@@ -233,3 +251,52 @@ test("Given a pool with 2 users with activated rewards WHEN more than one period
...
+    Assets.DEFAULT_AMOUNT.divn(2)
+  );
+
+  await waitForRewards(testUser1, liqIdPromPool2);
+
+  await promotePool(sudo.keyRingPair, liqIdPromPool2, 0);
+
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+
+  await claimRewardsAll(testUser1, liqIdPromPool2).then((result) => {
```
Comment (goncer) [inline test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts:268]: use regular claim

File: test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts [hunk 3]
```diff
@@ -233,3 +251,52 @@ test("Given a pool with 2 users with activated rewards WHEN more than one period
   expect(differenceMGAUser1).bnGt(BN_ZERO);
   expect(differenceMGAUser2).bnGt(BN_ZERO);
 });
+
+test("Given a pool with user with activated rewards  WHEN it was deactivated AND activated again THEN the user can receive a reward in both a deactivated and activated pool", async () => {
```
Comment (goncer) [inline test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts:255]: Lets use this test! :muscle:

File: test/parallel/xyk-pallet.API.pool.weightedPools.test.ts [hunk 4]
```diff
@@ -187,11 +187,21 @@ test("GIVEN a pool WHEN it has configured with 0 THEN no new issuance will be re
 
   await promotePool(sudo.keyRingPair, liqId, 0);
 
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
```
Comment (goncer) [inline test/parallel/xyk-pallet.API.pool.weightedPools.test.ts:190]: Lets use an specific test for it.

**Review by goncer — CHANGES_REQUESTED** [review]

File: test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts (hunk 1 above)
Comment (AlexChetverov) [reply test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts:287]: Merged to main branch. Fixed

File: test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts (hunk 2 above)
Comment (AlexChetverov) [reply test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts:268]: Fixed

File: test/parallel/xyk-pallet.API.pool.weightedPools.test.ts (hunk 4 above)
Comment (AlexChetverov) [reply test/parallel/xyk-pallet.API.pool.weightedPools.test.ts:190]: It looks like issue was fixed

Comment (AlexChetverov) [reply test/parallel/xyk-pallet.API.pool.weightedPools.test.ts]: Ok, chaged

File: test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts (hunk 3 above)
Comment (AlexChetverov) [reply test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts:255]: Done

File: test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts [hunk 5]
```diff
@@ -61,20 +67,21 @@ beforeAll(async () => {
 
   await setupApi();
 
-  [token1, token2] = await Assets.setupUserWithCurrencies(
+  [token1, token2, token3] = await Assets.setupUserWithCurrencies(
     sudo,
-    [defaultCurrencyValue, defaultCurrencyValue],
+    [defaultCurrencyValue, defaultCurrencyValue, defaultCurrencyValue],
```
Comment (goncer) [inline test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts:72]: :tada:

File: test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts [hunk 6]
```diff
@@ -61,20 +67,21 @@ beforeAll(async () => {
 
   await setupApi();
 
-  [token1, token2] = await Assets.setupUserWithCurrencies(
+  [token1, token2, token3] = await Assets.setupUserWithCurrencies(
```
Comment (goncer) [inline test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts:70]: plz, review that token2 and token3 are used because I think they are not.

**Review by goncer — APPROVED** [review]

File: test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts (hunk 6 above)
Comment (AlexChetverov) [reply test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts:70]: Yes, they used
image
image
image

**Pull request merged**